src/main/groovy/com/ullink/gradle/opencover/OpenCover.groovy

```diff
@@ -80,10 +109,48 @@ class OpenCover extends ConventionTask {
         if (skipAutoProps) commandLineArgs += '-skipautoprops'
         if (hideSkipped) commandLineArgs += '-hideskipped:' + hideSkipped
 
-        commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
+        commandLineArgs += "-target:${getTargetExec()}"
+        commandLineArgs += "-targetdir:${project.buildDir}"
+
+        commandLineArgs
+    }
+
+    def runSingleOpenCover(ArrayList commandLineArgs) {
+        commandLineArgs += ["\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\""]
         def filters = getTargetAssemblies().collect { "+[${FilenameUtils.getBaseName(project.file(it).name)}]*" }
         commandLineArgs += '-filter:\\"' + filters.join(' ') + '\\"'
         commandLineArgs += "-output:${getCoverageReportPath()}"
+
+        execute(commandLineArgs)
+    }
+
+    def runMultipleOpenCovers(ArrayList commandLineArgs) {
+        def intermediateReportsPath = new File(reportsFolder, "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
+
+        def nunitIntermediateFilesPath = new File(getIntermediateNunitResultsPath(), "")
+        nunitIntermediateFilesPath.mkdirs()
+
+        GParsPool.withPool {
+            getParallelTargetExecArgs().eachParallel {
+                def fileName = getRandomFileName()
+                logger.info("Filename generated for the ${it} input was ${fileName}")
+
+                commandLineArgs += ["\"-targetargs:${it.collect({escapeArg(it)}).join(' ')}\""]
+                commandLineArgs += "-output:${new File(intermediateReportsPath, fileName + ".xml")}"
+
+                execute(commandLineArgs)
+            }
+        }
+
+        execute(getMergeCommand(intermediateReportsPath))
+
+        new File(reportsFolder, "Summary.xml").renameTo(new File(reportsFolder, "coverage.xml"))
+        new File(intermediateReportsPath, "").deleteDir()
```

Comment:
intermediateReportsPath is already a `File`, so this can just be `intermediateReportsPath.deleteDir()` AFAICT

Reply:
Done